windows/src/developer/TIKE/xml/layoutbuilder/builder.js

```diff
@@ -1446,11 +1446,18 @@ $(function () {
     builder.updateKeyId(builder.selectedKey());
   }, {saveOnce: true});
 
+  const wrapInstant = function(f) {
+    return function() {
+      window.setTimeout(f.bind(this), 0);
+    }
+  }
+
```

Comment:
So what is this doing? avoiding a race condition?

Reply:
This is needed because the autocomplete `select` event happens before the text is changed in the target element, to give the callback a chance to block the selection. By adding a `setTimeout`, this delays the update to the next tick. The next tick is essentially immediate, but will always happen after the currently executing code returns control to the JavaScript event loop.